Trib 85

loretdemolas opened the pull request.

loretdemolas committed:
TRIB-84: Implement notifications api:

notifications.api.service.ts:

Created a mock api that simulates data. It takes two arrays, one for item and the other for message. It builds the notifications with the correct icon:message pairing.

notifications.page.html:

Added an ion List that uses a for loop to create list items. Each item has an onlick event listern to remove the notification from the notification array and display. If no notifications are displayed, it will display "no new notificaion"

notifications.page.ts:

Added logic to handleDelete of the notificaion.

Added a function to get notification icon and display it first in each list.

Added oninit lifecycle to call the api service and get notitications.

loretdemolas committed:
TRIB-84: Implement notifications api pt2

long-press.directive.ts:
Created a directive that use mousedown, mouseleave, mouseup, and timout to implement a longpress click.

notifications.page.module.ts:
importated and declared directive

notifications.page.html:
Added longpress event listenter to call onDelete function.
Added click event listerner to consol log more information on single click
Added CSS class to give user feedback when longpressing.

notifications.page.scss:
Added CSS to change ionitem color. Currently doesn't work. For some reason, animations break the longpress directive.

notifications.page.ts:
Added showMore function to console.log more mock information on each notification.

loretdemolas committed:
Merge pull request #1 from loretdemolas/develop

Merging latest develop changes into feature.

loretdemolas committed:
notifications.api.service.ts
-commented out deletenotification due to errors. It would not let the frontend load due to removal of mock api
-Modified getMessages to use backend endpoint to get notifications
notifications.page.html
-modified conditional statement to use async pipe to wait for notification instead of measuring the mockapi array length
notifications.page.ts
-Silenced onDeleteNotification beause there is no notifcation due to removal of mockapi
-added getUserID to find current users ID

loretdemolas committed:
Merge branch 'feature/notifications-page' into TRIB-85

haxwell commented:
As we talked about in Thursdays meeting, you need to implement a Model and an API service, in order for this code to be consistent with the rest of the app. Thanks.

loretdemolas committed:
_service/notifications.api.service.ts
renamed class file to follow code style
_service/notifications.model.service.ts
created model service to carry data
notifications.page.html
modified html to use the new model
notifications.page.ts
Modified getAllNotificationForUser to use model

loretdemolas committed:
notifications.model.service.ts
-changed the init to an async init to simplify code and await the response from api
-changed model name to notifications to improve readability

notifications.page.html
-I was using the async pipe when i should of measured the array length. Aysnc pipe for some reason did not work anymore.
notifications.page.ts
-Discovered the getter to get the value from model service.

made beautiful clean code.

loretdemolas commented:
Do not get confused about the simplicity of the code. This was a mountain! I was basically walking the data bit by bit but no matter what I  did I could not make the data jump from the model service to the component. 

*ENTERS THE GETTER*

With just two lines of code, this bad boy solved all my problems. I hate coding but damn do I want to learn more so I can one day be in the position to pass this golden method onto the next generation! I hope this is good code because I like how simple this method made this bit.

haxwell commented on src/app/pages/notifications-page/notifications.page.html:
```diff
@@ -1,18 +1,23 @@
+      <ion-item *ngFor="let notification of notifications"
+                appLongPress (longPress)="onDeleteNotification(notification.id)"
+                (click)="onShowMoreInfo(notification)"
+                [class.long-press]="isPressed">
```
Where is `isPressed` defined?

loretdemolas replied:
isPressed is like isClicked or onSubmit. Its a built in event listener. I am not sure if it is exclusive to angular.

haxwell commented on src/app/pages/notifications-page/_service/notifications.api.service.ts:
```diff
@@ -1,61 +1,31 @@
+        
+        const url = environment.apiUrl + '/api/notifications/user/' + this._authService.getUser().id;
+        
+        return new Promise((resolve, reject) => {
```
You could use async/await here, like you did in the model service.

haxwell commented:
Great work, I'm glad you put in the effort to [grok](https://www.google.com/search?q=grok) the pattern. I added a couple minor comments.

loretdemolas committed:
notifications.page.html
-Removed ion header and toolbar
notifications.page.ts
-Defined page header

haxwell commented:
Just as an update, there is a comment in [notifications.api.service.ts] re: await/async that is to be addressed.

loretdemolas committed:
_service/notifications.api.service.ts
-changed promise to an async await
/_service/notifications.model.service.ts
-added space to the end of the code
notifications.page.ts
-added space to end of the code
